programs/extcuraengine.hs

```diff
@@ -300,23 +299,28 @@ sliceLayer (Printer _ _ extruder) print@(Print _ infill lh _ _ ls outerWallBefor
           , drawInfill
           ]
         where
-          renderChildOuterContours src dest
-            | null childContours = travelBetweenContours src dest
-            | otherwise          = concat [
-                travelBetweenContours src $ head childContours
-                , drawOuterContour $ head childContours
-                , drawChildContours
-                , travelBetweenContours (last childContours) dest
-                ]
-          renderChildInnerContours src dest
-            | null childContoursInnerWalls = travelBetweenContours src dest
-            | otherwise          = concat [
-                travelBetweenContours src $ head childContoursInnerWalls
-                , drawInnerContour $ head childContoursInnerWalls
-                , drawChildOuterContours
-                , travelBetweenContours (last childContoursInnerWalls) dest
-                ]
-          -- Fail to the old contour shrink method when the skeleton based one knows it's failed.
+          renderChildOuterContours src dest = case childContours of
+            [] -> travelBetweenContours src dest
+            [headContour] ->
+              travelBetweenContours src headContour
+              <> drawOuterContour headContour
+              <> travelBetweenContours headContour dest
```

Comment:
This case is redundant; the next one covers it.  But I'm not sure which version is easier to read.  Perhaps you shouldn't change it.  (Same below.)

Reply:
yep. intentionally redundant, so the reader can work their brain up to the complicated case.